Protecs/src/test/java/uk/ac/ucl/protecs/sim/WorkplaceTesting.java

```diff
@@ -0,0 +1,220 @@
+	// ===== Here we test that the model is reading in workplaces from the census csv file and are being stored as workplace objects. =
+	// ===== We check that subsequently the workplace bubbles are created and are associated with the workplace location created. =====
+	// ===== We check that the bubbles are then created for each person, and that those who share a workplace have everyone associated=
+	// ===== with that workplace has been included in the bubble.                                                               =======
+	// ===== We check that people travel to their workplace location in the model.                                              =======
+	// ===== We check that the parameters being used to predict workplace contacts are being loaded                             =======
+	// ================================================================================================================================
+	private final String params;
+	
+	public WorkplaceTesting(String fileName) {
+		this.params = fileName;
+	}
+
+	@Test
+	public void checkPopulationWorkplacesAreBeingLoaded() {
+		// workplaces are read in from the csv file now, initialise the simulation
+		WorldBankCovid19Sim sim = helperFunctions.CreateDummySim("src/main/resources/workplace_bubbles_params.txt");
+		sim.start();
+		// check that the workplace locations have been created
+		Assert.assertTrue(sim.workplaces.size() > 0);
+
+	}
+	@Test
+	public void checkWorkplaceBubblesAreBeingMade() {
+		// bubbles are created after during the loading in process of the population
+		WorldBankCovid19Sim sim = helperFunctions.CreateDummySim("src/main/resources/workplace_bubbles_params.txt");
+		sim.start();
+		// check that everyone has a bubble associated with their workplace
+		boolean hasBeenGivenABubble = true;
+		for (Person p: sim.agents) {
+			if (p.getWorkBubble().isEmpty()) {hasBeenGivenABubble = false;}
+		}
+		Assert.assertTrue(hasBeenGivenABubble);
+	}
+	@Test
+	public void checkWorkplaceBubblesContainEveryoneInWorkplace() {
+		// bubbles are created after during the loading in process of the population
+
+		WorldBankCovid19Sim sim = helperFunctions.CreateDummySim("src/main/resources/workplace_bubbles_params.txt");
+		sim.start();
+		// create a function to search through the population and get people belonging to certain bubbles
+		Map<String, List<Person>> belongingToBubble = sim.agents.stream().collect(
+				Collectors.groupingBy(
+						Person::checkWorkplaceID
+						)
+				);
+		boolean bubbleContainsEveryoneInWorkplace = true;
+		// check that for each person, everyone who belongs to the workplace features in their bubble
+		for (Workplace w: sim.workplaces) {
+			String workplaceID = w.getId();
+			List<Person> thisBubble = belongingToBubble.get(workplaceID);
+			HashSet<Person> bubbleAsHashset = new HashSet<Person>(thisBubble);
+			for (Person p: thisBubble) {
+				bubbleContainsEveryoneInWorkplace = p.getWorkBubble().containsAll(bubbleAsHashset);
```

Comment:
let's keep an eye out of performance issues here as this'll be a heavy lift in terms of memory!